sdk/data/azcosmos/cosmos_client_retry_policy.go

```diff
@@ -100,7 +91,7 @@ func (p *clientRetryPolicy) attemptRetryOnEndpointFailure(req *policy.Request, i
 	} else {
 		p.gem.MarkEndpointUnavailableForRead(*req.Raw().URL)
 	}
-	p.gem.Update(req.Raw().Context())
+	p.gem.Update(req.Raw().Context(), true)
```

Comment:
For 403/1008, we don't set it to true I think https://github.com/Azure/azure-cosmos-dotnet-v3/blob/deab135c9b8709abf3e62f7be08e6c022513e4bf/Microsoft.Azure.Cosmos/src/ClientRetryPolicy.cs#L256

Reply:
we seem to do it for both Java and Python for the entire scenario - but I can make it so we only do it on 403.3
<img width="658" alt="image" src="https://github.com/Azure/azure-sdk-for-go/assets/30335873/47523042-6ac5-4e7f-b925-3e281c2d63bd">

<img width="588" alt="image" src="https://github.com/Azure/azure-sdk-for-go/assets/30335873/407f4785-8c50-4938-9c27-8950a5bc0fca">